### PR TITLE
Custom Library support in Spark pool

### PR DESCRIPTION
**Pull request by navneet35**
This is to select libraries to be included in the spark pool.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json [hunk 1]
```diff
@@ -410,6 +410,10 @@
           "type": "boolean",
           "description": "Whether compute isolation is required or not."
         },
+        "sessionLevelPackagesEnabled": {
+          "type": "boolean",
+          "description": "Whether session support is required on library or not."
```
Comment (maha-arun) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json:415]: "Whether session level package management is enabled or not."

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json [hunk 2]
```diff
@@ -502,6 +513,33 @@
         }
       }
     },
+    "LibraryInfo": {
+      "type": "object",
+      "description": "Library information of a Big Data pool powered by Apache Spark",
+      "title": "Spark Pool library information",
```
Comment (maha-arun) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json:519]: Information about a library/package created at the workspace level.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json [hunk 3]
```diff
@@ -502,6 +513,33 @@
         }
       }
     },
+    "LibraryInfo": {
+      "type": "object",
+      "description": "Library information of a Big Data pool powered by Apache Spark",
```
Comment (maha-arun) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json:518]: Library/package information.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json (hunk 2 above)
Comment (maha-arun) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json:519]: Library/package information

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json [hunk 4]
```diff
@@ -502,6 +513,33 @@
...
+          "type": "string",
+          "description": "Storage blob path of library."
+        },
+        "containerName": {
+          "type": "string",
+          "description": "Storage blob container name."
+        },
+        "uploadedTimestamp": {
+          "type": "datetimeoffset",
+          "description": "The last update time of the library.."
```
Comment (maha-arun) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json:535]: extra . at the end

**Comment by raych1** [comment]
@navneet35 , please fix CI check failures.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json (hunk 2 above)
Comment (navneet35) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json]: added

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json (hunk 4 above)
Comment (navneet35) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json]: removed the extra (.)

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json (hunk 3 above)
Comment (navneet35) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json]: added

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json (hunk 2 above)
Comment (navneet35) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json]: changed

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json [hunk 5]
```diff
@@ -423,6 +427,13 @@
           "$ref": "#/definitions/LibraryRequirements",
           "description": "Library version requirements"
         },
+        "customLibraries": {
+          "description": "Custom libraries to be included in spark pool",
```
[the remark this answers is not in the document]
Comment (navneet35) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json]: changed the description

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json (hunk 1 above)
Comment (navneet35) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json]: changed

**Comment by idear1203** [comment]
@raych1 , Synapse service is still in preview. According to the doc Section 4, preview services are not in the scope of the policy.
> Please note the following are currently not in scope of the policy: Preview services, non-Azure service
changes; Maintenance updates, Email communication- template changes.

**Review by raych1 — APPROVED** [review]

**Comment by raych1** [comment]
Approved the breaking change as this service is not GA yet.

**Pull request merged**